### PR TITLE
WT-2309 Add option to cause delays in internal page split code to aid testing

### PR DESCRIPTION
**Pull request by sulabhM**

### CODE REVIEW

**Comment by sulabhM** [comment]
@agorrod , please have a look

**Comment by agorrod** [comment]
@sulabhM the option name lgtm

**Comment by sulabhM** [comment]
@bvpvamsikrishna, can you please review this change.

File: dist/flags.py [hunk 1]
```diff
@@ -48,6 +48,7 @@
     'timing_stress_for_test' : [
         'TIMING_STRESS_CHECKPOINT_SLOW',
         'TIMING_STRESS_PAGE_SPLIT_RACE',
+        'TIMING_STRESS_INTERNAL_PAGE_SPLIT_RACE',
```
Comment (bvpvamsikrishna) [inline dist/flags.py]: All the parameters to be in sorted order.

File: dist/api_data.py [hunk 2]
```diff
@@ -526,9 +526,9 @@ def __cmp__(self, other):
         This option is intended for use with internal stress
         testing of WiredTiger. Options are given as a list, such as
         <code>"timing_stress_for_test=[checkpoint_slow,
-            page_split_race]"</code>''',
+            page_split_race, internal_page_split_race]"</code>''',
```
Comment (bvpvamsikrishna) [inline dist/api_data.py]: Arrange this also as per below order

**Comment by sulabhM** [comment]
@keithbostic , please have a look at this change. I based this one on an earlier change that you made:
https://github.com/wiredtiger/wiredtiger/pull/2424

Let me know in case adding additional delays will help.

File: src/btree/bt_split.c [hunk 3]
```diff
@@ -1061,9 +1069,18 @@ __split_internal(WT_SESSION_IMPL *session, WT_PAGE *parent, WT_PAGE *page)
...
+	    WT_TIMING_STRESS_INTERNAL_PAGE_SPLIT_RACE, 100 * WT_THOUSAND);
+
 	/* Split into the parent. */
 	if ((ret = __split_parent(session, page_ref, alloc_index->index,
 	    alloc_index->entries, parent_incr, false, false)) == 0) {
+		/* Encourage a race */
+		__page_split_timing_stress(session,
+		    WT_TIMING_STRESS_INTERNAL_PAGE_SPLIT_RACE,
+		    100 * WT_THOUSAND);
+
```
Comment (keithbostic) [inline src/btree/bt_split.c]: I'd suggest moving this stress test into the `__split_parent()` function, that way we'll test other parent split code as well as this path (for example, `__wt_split_reverse()` and `__split_insert()` call `__split_parent()` directly.

**Comment by agorrod** [comment]
Thanks @sulabhM lgtm. Let's get this merged into develop, then merge develop up to #3479.

@keithbostic If you have suggestions for moving around the yield calls we can make those changes in #3479.

**Pull request merged**